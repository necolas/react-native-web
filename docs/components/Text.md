# Text

`Text` is component for displaying text. It supports style, basic touch
handling, and inherits typographic styles from ancestor elements.

The `Text` is unique relative to layout: child elements use text layout
(`inline`) rather than flexbox layout. This means that elements inside of a
`Text` are not rectangles, as they wrap when reaching the edge of their
container.

Unsupported React Native props:
`allowFontScaling` (ios),
`suppressHighlighting` (ios)

## Props

NOTE: `Text` will transfer all other props to the rendered HTML element.

(web) **accessibilityLabel**: string

Defines the text available to assistive technologies upon interaction with the
element. (This is implemented using `aria-label`.)

(web) **accessibilityRole**: oneOf(roles)

Allows assistive technologies to present and support interaction with the view
in a manner that is consistent with user expectations for similar views of that
type. For example, marking a touchable view with an `accessibilityRole` of
`button`. (This is implemented using [ARIA roles](http://www.w3.org/TR/wai-aria/roles#role_definitions)).

Note: Avoid changing `accessibilityRole` values over time or after user
actions. Generally, accessibility APIs do not provide a means of notifying
assistive technologies of a `role` value change.

(web) **accessible**: bool = true

When `false`, the text is hidden from assistive technologies. (This is
implemented using `aria-hidden`.)

**children**: any

Child content.

**numberOfLines**: number

Truncates the text with an ellipsis after this many lines. Currently only supports `1`.

**onPress**: function

This function is called on press.

**style**: style

+ ...[View#style](View.md)
+ `color`
+ `fontFamily`
+ `fontSize`
+ `fontStyle`
+ `fontWeight`
+ `letterSpacing`
+ `lineHeight`
+ `textAlign`
+ `textAlignVertical`
+ `textDecorationLine`
+ `textShadow`
+ `textTransform`
+ `whiteSpace`
+ `wordWrap`
+ `writingDirection`

**testID**: string

Used to locate this view in end-to-end tests.

## Examples

```js
import React, { Component, PropTypes, StyleSheet, Text } from 'react-native'

export default class PrettyText extends Component {
  static propTypes = {
    ...Text.propTypes,
    color: PropTypes.oneOf(['white', 'gray', 'red']),
    size: PropTypes.oneOf(['small', 'normal', 'large']),
    weight: PropTypes.oneOf(['light', 'normal', 'bold'])
  };

  static defaultProps = {
    ...Text.defaultProps,
    color: 'gray',
    size: 'normal',
    weight: 'normal'
  };

  render() {
    const { color, size, style, weight, ...other } = this.props;

    return (
      <Text
        ...other
        style={[
          style,
          colorStyles[color],
          sizeStyles[size],
          weightStyles[weight]
        ]}
      />
    );
  }
}

const colorStyles = StyleSheet.create({
  white: { color: 'white' },
  gray: { color: 'gray' },
  red: { color: 'red' }
})

const sizeStyles = StyleSheet.create({
  small: { fontSize: '0.85rem', padding: '0.5rem' },
  normal: { fontSize: '1rem', padding: '0.75rem' },
  large: { fontSize: '1.5rem', padding: '1rem' }
})

const weightStyles = StyleSheet.create({
  light: { fontWeight: '300' },
  normal: { fontWeight: '400' },
  bold: { fontWeight: '700' }
})
```
