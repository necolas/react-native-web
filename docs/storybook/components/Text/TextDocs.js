/* eslint-disable react/jsx-sort-props */

/**
 * @flow
 */

import PropChildren from './examples/PropChildren';
import PropNumberOfLines from './examples/PropNumberOfLines';
import PropOnPress from './examples/PropOnPress';
import React from 'react';
import { storiesOf } from '@kadira/storybook';
import UIExplorer, { AppText, Code, DocItem, StyleList } from '../../ui-explorer';

const stylePropTypes = [
  {
    name: '...View#style'
  },
  {
    name: 'color',
    typeInfo: 'color'
  },
  {
    name: 'fontFamily',
    typeInfo: 'string'
  },
  {
    label: 'web',
    name: 'fontFeatureSettings',
    typeInfo: 'string'
  },
  {
    name: 'fontSize',
    typeInfo: 'number | string'
  },
  {
    name: 'fontStyle',
    typeInfo: 'string'
  },
  {
    name: 'fontWeight',
    typeInfo: 'string'
  },
  {
    name: 'letterSpacing',
    typeInfo: 'string'
  },
  {
    name: 'lineHeight',
    typeInfo: 'string'
  },
  {
    name: 'textAlign',
    typeInfo: 'string'
  },
  {
    name: 'textAlignVertical',
    typeInfo: 'string'
  },
  {
    name: 'textDecorationLine',
    typeInfo: 'string'
  },
  {
    label: 'web',
    name: 'textIndent',
    typeInfo: 'string'
  },
  {
    label: 'web',
    name: 'textOverflow',
    typeInfo: 'string'
  },
  {
    label: 'web',
    name: 'textRendering',
    typeInfo: 'string'
  },
  {
    name: 'textShadowColor',
    typeInfo: 'color'
  },
  {
    name: 'textShadowOffset',
    typeInfo: '{ height: number, width: number }'
  },
  {
    name: 'textShadowRadius',
    typeInfo: 'number | string'
  },
  {
    label: 'web',
    name: 'textTransform',
    typeInfo: 'string'
  },
  {
    label: 'web',
    name: 'unicodeBidi',
    typeInfo: 'string'
  },
  {
    name: 'whiteSpace',
    typeInfo: 'string'
  },
  {
    label: 'web',
    name: 'wordWrap',
    typeInfo: 'string'
  },
  {
    name: 'writingDirection',
    typeInfo: 'string'
  }
];

const sections = [
  {
    title: 'Props',
    entries: [
      <DocItem
        name="accessibilityLabel"
        typeInfo="?string"
        description="Overrides the text that is read by a screen reader when the user interacts with the element. (This is implemented using 'aria-label'.)"
      />,

      <DocItem
        name="accessibilityLiveRegion"
        typeInfo="?enum('assertive', 'none', 'polite')"
        description={
          <AppText>
            Indicates to assistive technologies whether to notify the user when the view changes.
            The values of
            this attribute are expressed in degrees of importance. When regions are specified as{
              ' '
            }
            <Code>polite</Code> (recommended),
            updates take low priority. When regions are specified as <Code>assertive</Code>,
            assistive technologies will
            interrupt and immediately notify the user. (This is implemented using 'aria-live'.)
          </AppText>
        }
      />,

      <DocItem
        label="web"
        name="accessibilityRole"
        typeInfo="?enum(roles)"
        description={
          <AppText>
            Allows assistive technologies to present and support interaction with the view in a
            manner that is
            consistent with user expectations for similar views of that type. For example, marking a
            touchable
            view with an <Code>accessibilityRole</Code> of <Code>button</Code>. For compatibility
            with React
            Native <Code>accessibilityTraits</Code> and <Code>accessibilityComponentType</Code> are
            mapped
            to <Code>accessibilityRole</Code>. (This is implemented using ARIA roles.)
          </AppText>
        }
      />,

      <DocItem
        name="accessible"
        typeInfo="?boolean"
        description={
          <AppText>
            When <Code>true</Code>, indicates that the view is an accessibility element (i.e.,
            focusable) and
            groups its child content. By default, all the touchable elements and elements
            with <Code>accessibilityRole</Code> of <Code>button</Code> and <Code>link</Code> are
            accessible.
            (This is implemented using 'tabindex'.)
          </AppText>
        }
      />,

      <DocItem
        name="children"
        typeInfo="?any"
        description={`Child content. Nested text components will inherit the styles of their parents
(only backgroundColor is inherited from non-Text parents). <Text>
only supports other <Text> and raw text (strings) as children.`}
        example={{
          code: '',
          render: () => <PropChildren />
        }}
      />,

      <DocItem
        name="importantForAccessibility"
        typeInfo="?enum('auto', 'no', 'no-hide-descendants', 'yes')"
        description={
          'A value of `no` will remove the element from the tab flow.\n\nA value of `no-hide-descendants` will hide the element and its children from assistive technologies. (This is implemented using `aria-hidden`.)'
        }
      />,

      <DocItem
        name="numberOfLines"
        typeInfo="?number"
        description="Truncates the text with an ellipsis after this many lines. Currently only supports `1`."
        example={{
          code: '',
          render: () => <PropNumberOfLines />
        }}
      />,

      <DocItem
        name="onLayout"
        typeInfo="?function"
        description="Invoked on mount and layout changes with `{ nativeEvent: { layout: { x, y, width, height } } }`, where `x` and `y` are the offsets from the parent node."
      />,

      <DocItem
        name="onPress"
        typeInfo="?function"
        description="Called when the Text is pressed"
        example={{
          code: '',
          render: () => <PropOnPress />
        }}
      />,

      <DocItem
        name="selectable"
        typeInfo="?boolean"
        description="When `false`, the text is not selectable."
      />,

      <DocItem
        name="style"
        typeInfo="?style"
        description={<StyleList stylePropTypes={stylePropTypes} />}
      />,

      <DocItem
        name="testID"
        typeInfo="?string"
        description="Used to locate this view in end-to-end tests. The test ID is rendered to a `data-testid` DOM attribute"
      />
    ]
  }
];

storiesOf('Components', module).add('Text', () =>
  <UIExplorer
    description={[
      <AppText>
        Text is component for displaying text. It supports style, basic touch handling, and inherits
        typographic
        styles from ancestor elements.
      </AppText>,
      <AppText>
        Text is unique relative to layout: child elements use text layout ("inline") rather than
        flexbox layout.
        This means that elements inside of a Text are not rectangles, as they wrap when reaching the
        edge of
        their container.
      </AppText>,
      <AppText>NOTE: Text will transfer all other props to the rendered HTML element.</AppText>
    ]}
    sections={sections}
    title="Text"
    url="components/Text"
  />
);
