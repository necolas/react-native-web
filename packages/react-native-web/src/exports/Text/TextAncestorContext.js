/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict
 */

import type { Context } from 'react';

import { createContext } from 'react';

const TextAncestorContext = createContext(false);
export default (TextAncestorContext: Context<boolean>);
