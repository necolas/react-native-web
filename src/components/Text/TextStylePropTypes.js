/**
 * @flow
 */

import ColorPropType from '../../propTypes/ColorPropType';
import ViewStylePropTypes from '../View/ViewStylePropTypes';
import { number, oneOf, oneOfType, shape, string } from 'prop-types';

const numberOrString = oneOfType([number, string]);

const ShadowOffsetPropType = shape({ width: number, height: number });
const TextAlignPropType = oneOf(['center', 'inherit', 'justify', 'justify-all', 'left', 'right']);
const WritingDirectionPropType = oneOf(['auto', 'ltr', 'rtl']);

const TextStylePropTypes = {
  ...ViewStylePropTypes,
  color: ColorPropType,
  fontFamily: string,
  fontFeatureSettings: string,
  fontSize: numberOrString,
  fontStyle: string,
  fontWeight: string,
  letterSpacing: numberOrString,
  lineHeight: numberOrString,
  textAlign: TextAlignPropType,
  textAlignVertical: oneOf(['auto', 'bottom', 'center', 'top']),
  textDecorationColor: ColorPropType,
  textDecorationLine: string,
  textDecorationStyle: string,
  textShadowColor: ColorPropType,
  textShadowOffset: ShadowOffsetPropType,
  textShadowRadius: number,
  writingDirection: WritingDirectionPropType,
  /* @platform web */
  textIndent: numberOrString,
  textOverflow: string,
  textRendering: oneOf(['auto', 'geometricPrecision', 'optimizeLegibility', 'optimizeSpeed']),
  textTransform: oneOf(['capitalize', 'lowercase', 'none', 'uppercase']),
  unicodeBidi: oneOf([
    'normal',
    'bidi-override',
    'embed',
    'isolate',
    'isolate-override',
    'plaintext'
  ]),
  whiteSpace: string,
  wordWrap: string,
  MozOsxFontSmoothing: string,
  WebkitFontSmoothing: string
};

export default TextStylePropTypes;
