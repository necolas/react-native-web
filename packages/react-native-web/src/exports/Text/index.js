/**
 * Copyright (c) Nicolas Gallagher.
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

import type { PlatformMethods } from '../../types';
import type { TextProps } from './types';

import * as React from 'react';
import createElement from '../createElement';
import * as forwardedProps from '../../modules/forwardedProps';
import pick from '../../modules/pick';
import useElementLayout from '../../modules/useElementLayout';
import useMergeRefs from '../../modules/useMergeRefs';
import usePlatformMethods from '../../modules/usePlatformMethods';
import useResponderEvents from '../../modules/useResponderEvents';
import StyleSheet from '../StyleSheet';
import TextAncestorContext from './TextAncestorContext';
import { useLocaleContext, getLocaleDirection } from '../../modules/useLocale';

const forwardPropsList = Object.assign(
  {},
  forwardedProps.defaultProps,
  forwardedProps.accessibilityProps,
  forwardedProps.clickProps,
  forwardedProps.focusProps,
  forwardedProps.keyboardProps,
  forwardedProps.mouseProps,
  forwardedProps.touchProps,
  forwardedProps.styleProps,
  {
    href: true,
    lang: true,
    pointerEvents: true
  }
);

const pickProps = (props) => pick(props, forwardPropsList);

const Text: React.AbstractComponent<TextProps, HTMLElement & PlatformMethods> =
  React.forwardRef((props, forwardedRef) => {
    const {
      hrefAttrs,
      numberOfLines,
      onClick,
      onLayout,
      onPress,
      onMoveShouldSetResponder,
      onMoveShouldSetResponderCapture,
      onResponderEnd,
      onResponderGrant,
      onResponderMove,
      onResponderReject,
      onResponderRelease,
      onResponderStart,
      onResponderTerminate,
      onResponderTerminationRequest,
      onScrollShouldSetResponder,
      onScrollShouldSetResponderCapture,
      onSelectionChangeShouldSetResponder,
      onSelectionChangeShouldSetResponderCapture,
      onStartShouldSetResponder,
      onStartShouldSetResponderCapture,
      selectable,
      ...rest
    } = props;

    const hasTextAncestor = React.useContext(TextAncestorContext);
    const hostRef = React.useRef(null);
    const { direction: contextDirection } = useLocaleContext();

    useElementLayout(hostRef, onLayout);
    useResponderEvents(hostRef, {
      onMoveShouldSetResponder,
      onMoveShouldSetResponderCapture,
      onResponderEnd,
      onResponderGrant,
      onResponderMove,
      onResponderReject,
      onResponderRelease,
      onResponderStart,
      onResponderTerminate,
      onResponderTerminationRequest,
      onScrollShouldSetResponder,
      onScrollShouldSetResponderCapture,
      onSelectionChangeShouldSetResponder,
      onSelectionChangeShouldSetResponderCapture,
      onStartShouldSetResponder,
      onStartShouldSetResponderCapture
    });

    const handleClick = React.useCallback(
      (e) => {
        if (onClick != null) {
          onClick(e);
        } else if (onPress != null) {
          e.stopPropagation();
          onPress(e);
        }
      },
      [onClick, onPress]
    );

    let component = hasTextAncestor ? 'span' : 'div';

    const langDirection =
      props.lang != null ? getLocaleDirection(props.lang) : null;
    const componentDirection = props.dir || langDirection;
    const writingDirection = componentDirection || contextDirection;

    const supportedProps = pickProps(rest);
    supportedProps.dir = componentDirection;
    // 'auto' by default allows browsers to infer writing direction (root elements only)
    if (!hasTextAncestor) {
      supportedProps.dir =
        componentDirection != null ? componentDirection : 'auto';
    }

    if (onClick || onPress) {
      supportedProps.onClick = handleClick;
    }

    supportedProps.style = [
      numberOfLines != null &&
        numberOfLines > 1 && { WebkitLineClamp: numberOfLines },
      hasTextAncestor === true ? styles.textHasAncestor$raw : styles.text$raw,
      numberOfLines === 1 && styles.textOneLine,
      numberOfLines != null && numberOfLines > 1 && styles.textMultiLine,
      props.style,
      selectable === true && styles.selectable,
      selectable === false && styles.notSelectable,
      onPress && styles.pressable
    ];

    if (props.href != null) {
      component = 'a';
      if (hrefAttrs != null) {
        const { download, rel, target } = hrefAttrs;
        if (download != null) {
          supportedProps.download = download;
        }
        if (rel != null) {
          supportedProps.rel = rel;
        }
        if (typeof target === 'string') {
          supportedProps.target =
            target.charAt(0) !== '_' ? '_' + target : target;
        }
      }
    }

    const platformMethodsRef = usePlatformMethods(supportedProps);
    const setRef = useMergeRefs(hostRef, platformMethodsRef, forwardedRef);

    supportedProps.ref = setRef;

    const element = createElement(component, supportedProps, {
      writingDirection
    });

    return hasTextAncestor ? (
      element
    ) : (
      <TextAncestorContext.Provider value={true}>
        {element}
      </TextAncestorContext.Provider>
    );
  });

Text.displayName = 'Text';

const textStyle = {
  backgroundColor: 'transparent',
  border: '0 solid black',
  boxSizing: 'border-box',
  color: 'black',
  display: 'inline',
  font: '14px System',
  listStyle: 'none',
  margin: 0,
  padding: 0,
  textAlign: 'inherit',
  textDecoration: 'none',
  whiteSpace: 'pre-wrap',
  wordWrap: 'break-word'
};

const styles = StyleSheet.create({
  text$raw: textStyle,
  textHasAncestor$raw: {
    ...textStyle,
    color: 'inherit',
    font: 'inherit',
    whiteSpace: 'inherit'
  },
  textOneLine: {
    maxWidth: '100%',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap',
    wordWrap: 'normal'
  },
  // See #13
  textMultiLine: {
    display: '-webkit-box',
    maxWidth: '100%',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    WebkitBoxOrient: 'vertical'
  },
  notSelectable: {
    userSelect: 'none'
  },
  selectable: {
    userSelect: 'text'
  },
  pressable: {
    cursor: 'pointer'
  }
});

export default Text;
