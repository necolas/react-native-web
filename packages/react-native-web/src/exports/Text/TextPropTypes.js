/**
 * Copyright (c) 2015-present, Nicolas Gallagher.
 * Copyright (c) 2015-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @providesModule TextPropTypes
 * @flow
 */

import StyleSheetPropType from '../../modules/StyleSheetPropType';
import TextStylePropTypes from './TextStylePropTypes';
import { any, array, bool, func, number, oneOf, oneOfType, string } from 'prop-types';

const TextPropTypes = {
  accessibilityComponentType: string,
  accessibilityLabel: string,
  accessibilityLiveRegion: oneOf(['assertive', 'none', 'polite']),
  accessibilityRole: oneOf(['button', 'heading', 'label', 'link', 'listitem']),
  accessibilityTraits: oneOfType([array, string]),
  accessible: bool,
  children: any,
  importantForAccessibility: oneOf(['auto', 'no', 'no-hide-descendants', 'yes']),
  numberOfLines: number,
  onBlur: func,
  onContextMenu: func,
  onFocus: func,
  onLayout: func,
  onPress: func,
  selectable: bool,
  style: StyleSheetPropType(TextStylePropTypes),
  testID: string
};

export default TextPropTypes;
