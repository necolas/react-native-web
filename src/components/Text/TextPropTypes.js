/**
 * Copyright (c) 2015-present, Nicolas Gallagher.
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @providesModule TextPropTypes
 * @flow
 */

import BaseComponentPropTypes from '../../propTypes/BaseComponentPropTypes';
import StyleSheetPropType from '../../propTypes/StyleSheetPropType';
import TextStylePropTypes from './TextStylePropTypes';
import { any, bool, func, number, oneOf } from 'prop-types';

const TextPropTypes = {
  ...BaseComponentPropTypes,
  accessibilityRole: oneOf(['button', 'heading', 'label', 'link', 'listitem']),
  children: any,
  numberOfLines: number,
  onLayout: func,
  onPress: func,
  selectable: bool,
  style: StyleSheetPropType(TextStylePropTypes)
};

export default TextPropTypes;
