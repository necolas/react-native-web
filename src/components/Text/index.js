import applyLayout from '../../modules/applyLayout';
import applyNativeMethods from '../../modules/applyNativeMethods';
import BaseComponentPropTypes from '../../propTypes/BaseComponentPropTypes';
import createDOMElement from '../../modules/createDOMElement';
import StyleSheet from '../../apis/StyleSheet';
import StyleSheetPropType from '../../propTypes/StyleSheetPropType';
import TextStylePropTypes from './TextStylePropTypes';
import { Component, PropTypes } from 'react';

class Text extends Component {
  static displayName = 'Text';

  static propTypes = {
    ...BaseComponentPropTypes,
    accessibilityRole: PropTypes.oneOf([ 'button', 'heading', 'link', 'listitem' ]),
    children: PropTypes.any,
    numberOfLines: PropTypes.number,
    onLayout: PropTypes.func,
    onPress: PropTypes.func,
    selectable: PropTypes.bool,
    style: StyleSheetPropType(TextStylePropTypes)
  };

  static defaultProps = {
    accessible: true,
    selectable: true
  };

  render() {
    const {
      numberOfLines,
      onPress,
      selectable,
      style,
      /* eslint-disable */
      adjustsFontSizeToFit,
      allowFontScaling,
      ellipsizeMode,
      minimumFontScale,
      onLayout,
      suppressHighlighting,
      /* eslint-enable */
      ...otherProps
    } = this.props;

    if (onPress) {
      otherProps.onClick = onPress;
      otherProps.onKeyDown = this._createEnterHandler(onPress);
      otherProps.tabIndex = 0;
    }

    otherProps.style = [
      styles.initial,
      style,
      !selectable && styles.notSelectable,
      numberOfLines === 1 && styles.singleLineStyle,
      onPress && styles.pressable
    ];
    // allow browsers to automatically infer the language writing direction
    otherProps.dir = 'auto';

    return createDOMElement('span', otherProps);
  }

  _createEnterHandler(fn) {
    return (e) => {
      if (e.keyCode === 13) {
        fn && fn(e);
      }
    };
  }
}

const styles = StyleSheet.create({
  initial: {
    borderWidth: 0,
    color: 'inherit',
    display: 'inline',
    font: 'inherit',
    margin: 0,
    padding: 0,
    textDecorationLine: 'none',
    whiteSpace: 'pre-wrap',
    wordWrap: 'break-word'
  },
  notSelectable: {
    userSelect: 'none'
  },
  pressable: {
    cursor: 'pointer'
  },
  singleLineStyle: {
    maxWidth: '100%',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap'
  }
});

module.exports = applyLayout(applyNativeMethods(Text));
