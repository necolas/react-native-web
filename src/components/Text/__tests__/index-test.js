/* eslint-env jasmine, jest */

import React from 'react';
import renderer from 'react-test-renderer';
import Text from '../';

jest.mock('react-dom');

describe('components/Text', () => {
  test('prop "children"', () => {
    const component = renderer.create(<Text>children</Text>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('prop "numberOfLines"');

  test('prop "onPress"', () => {
    const onPress = (e) => {};
    const component = renderer.create(<Text onPress={onPress} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('prop "selectable"', () => {
    let component = renderer.create(<Text />);
    expect(component.toJSON()).toMatchSnapshot();
    component = renderer.create(<Text selectable={false} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
