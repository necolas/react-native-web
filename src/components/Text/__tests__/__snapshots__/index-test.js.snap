exports[`components/Text prop "children" 1`] = `
<span
  className=""
  style={
    Object {
      "borderBottomWidth": "0px",
      "borderLeftWidth": "0px",
      "borderRightWidth": "0px",
      "borderTopWidth": "0px",
      "color": "inherit",
      "display": "inline",
      "font": "inherit",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "textDecoration": "none",
      "wordWrap": "break-word",
    }
  }>
  children
</span>
`;

exports[`components/Text prop "onPress" 1`] = `
<span
  className=""
  onClick={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "borderBottomWidth": "0px",
      "borderLeftWidth": "0px",
      "borderRightWidth": "0px",
      "borderTopWidth": "0px",
      "color": "inherit",
      "cursor": "pointer",
      "display": "inline",
      "font": "inherit",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "textDecoration": "none",
      "wordWrap": "break-word",
    }
  }
  tabIndex={0} />
`;

exports[`components/Text prop "selectable" 1`] = `
<span
  className=""
  style={
    Object {
      "borderBottomWidth": "0px",
      "borderLeftWidth": "0px",
      "borderRightWidth": "0px",
      "borderTopWidth": "0px",
      "color": "inherit",
      "display": "inline",
      "font": "inherit",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "textDecoration": "none",
      "wordWrap": "break-word",
    }
  } />
`;

exports[`components/Text prop "selectable" 2`] = `
<span
  className=""
  style={
    Object {
      "MozUserSelect": "none",
      "WebkitUserSelect": "none",
      "borderBottomWidth": "0px",
      "borderLeftWidth": "0px",
      "borderRightWidth": "0px",
      "borderTopWidth": "0px",
      "color": "inherit",
      "display": "inline",
      "font": "inherit",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "msUserSelect": "none",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "textDecoration": "none",
      "userSelect": "none",
      "wordWrap": "break-word",
    }
  } />
`;
